.github/workflows/codecov.yaml

```diff
@@ -0,0 +1,22 @@
+name: Code Coverage
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 2
+      - uses: actions/setup-go@v2
+        with:
+          go-version-file: "go.mod"
```

Comment:
`Unexpected input(s) 'go-version-file', valid inputs are ['go-version', 'check-latest', 'stable', 'token']`

This feat is added in https://github.com/actions/setup-go/releases/tag/v3.1.0 so, should change `actions/setup-go@v2` to `actions/setup-go@v3` or `actions/setup-go@v4` (latest)